public/javascripts/studentControllers.js

```diff
@@ -138,5 +143,59 @@ var addStudentController = function($scope, $http, $location) {
       $scope.studentAddedMsg = response.data.msg;
     });
 
-  };
-};
+  }
+}
+
+var studentDataController = function($scope, $http, $state) {
+
+  $scope.setType = function(dataType){
+    $scope.dataTypeSelected = dataType;
+    console.log('getDataEntries for', $scope.currentStudent.name);
+    $http.get('/api/student/data/' + $scope.currentStudent._id + '/' + dataType)
+      .then(function successCallback(response) {
+        console.log(response.data);
+        if(response.data[0] == undefined){
+          $scope.dateSelected = "No data available";
+          $scope.isData = false;
+        } else{
+          $scope.currentDateList = response.data;
+          $scope.dateSelected = response.data[0];
+          $scope.isData = true;
+        }
+      }, function errorCallback(response) {
+        console.log('Error: ' + response.data);
+    });
+  }
+
+  $scope.getDateString = function(dateString){
+    var date = new Date(dateString);
+    var days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+    var months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+    return days[date.getDay()] + ', ' + months[date.getMonth()] + ' ' + date.getDate(); 
```

Comment:
from what i saw i could only use the date object to get numbers - you can put wierd date formats into the object, but what you get out is numbers depending on the methods you use.